src/cogent3/util/io.py

```diff
@@ -343,8 +344,81 @@ def remove_files(list_of_filepaths, error_on_missing=True):
         raise OSError("Some filepaths were not accessible: %s" % "\t".join(missing))
 
 
-def path_exists(path: os.PathLike) -> bool:
+def path_exists(path: PathType) -> bool:
     """whether path is a valid path and it exists"""
-    with contextlib.suppress(ValueError, TypeError):
-        return os_path.exists(str(path))
+    with contextlib.suppress(Exception):
+        return Path(path).exists()
     return False
+
+
+def iter_splitlines(
+    path: PathType, chunk_size: Optional[int] = 1_000_000
+) -> Generator[str, None, None]:
+    """yields line from file
+
+    Parameters
+    ----------
+    path
+        data file
+    chunk_size
+        number of bytes to load in one go from path
+
+    Notes
+    -----
+    Loads chunks of data from the file, yields one line at a time
+    """
+    path = Path(path)
+    if chunk_size and path.stat().st_size < chunk_size:
+        # file is smaller than provided chunk_size, just
+        # load it all
+        chunk_size = None
+
+    with open_(path) as infile:
+        last = ""
+        while True:
+            data = infile.read(chunk_size)
+            if not data:  # end of file
+                break
+
+            data = last + data
+            # we check for POSIX or Windows line endings
+            end_is_newline = data.endswith(("\n", "\r\n"))
```

Comment:
Should be equivalent to just `data.endswith("\n")`

Reply:
if you read in all text from a text file that was created on Windows (thus "\r\n" endings), on Windows what line endings are in the text?

Reply:
It's complicated. Though in either case a line which ends with "\r\n" still ends with "\n" so there is redundancy.